### PR TITLE
refactor(tools): separate module-related functions from tool.utils

### PR DESCRIPTION
**Pull request by Water-Melon**
### Summary

Move module-related functions from `kong/tools/utils.lua` to `kong/tools/module.lua`.

### Issue reference

KAG-2960

### CODE REVIEW

File: kong/tools/module.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
+local find = string.find
```
Comment (chronolaw) [inline kong/tools/module.lua:1]: We should also localize `type` `error` `xpcall`

File: kong/tools/module.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+local _M = {}
+
+--- Try to load a module.
```
[the remark this answers is not in the document]
Comment (Water-Melon) [reply kong/tools/module.lua:11]: done

File: kong/tools/module.lua (hunk 1 above)
Comment (Water-Melon) [reply kong/tools/module.lua]: done

**Review by chronolaw — APPROVED** [review]

**Comment by fffonion** [comment]
this only involves one function so doesn't seem to worth seperate into a different file?

**Comment by Water-Melon** [comment]
> this only involves one function so doesn't seem to worth seperate into a different file?

The `kong/tools/utils.lua` file will be removed.

Currently, this process is divided into three steps:
- The first step involves splitting the content of `kong/tools/utils.lua' and categorizing it into various files
- The second step is to correct all external dependencies of each component.
- The last step is to remove `kong/tools/utils.lua'.

Therefore, it still needs to be separated.

**Comment by bungle** [comment]
> * The last step is to remove `kong/tools/utils.lua`.

Are we going to see final step too that moves things in PDK?

**Comment by chronolaw** [comment]
> > * The last step is to remove `kong/tools/utils.lua`.
> 
> Are we going to see the final step too that moves things in PDK? :-)

I think that pdk is another topic, refactoring of pdk may be more careful.

**Comment by bungle** [comment]
> > > * The last step is to remove `kong/tools/utils.lua`.
> > 
> > 
> > Are we going to see the final step too that moves things in PDK? :-)
> 
> I think that pdk is another topic, refactoring of pdk may be more careful.

Yes, there is then opportunity to think things better, test things better, document things better... but that should be the ultimate goal as these utils are most useful with plugin developers.

**Review by bungle — APPROVED** [review]

**Pull request merged**